### PR TITLE
sql: add test that shows fk observing txn's prior writes

### PR DESCRIPTION
**Pull request by dt**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/13950)

### CODE REVIEW

**Review by a-robinson — APPROVED** [review]

File: pkg/sql/testdata/fk [hunk 1]
```diff
@@ -526,3 +526,38 @@ CREATE TABLE domain_modules (
...
+COMMIT
+
+statement ok
+BEGIN TRANSACTION
+
+statement error foreign key violation: value \[3\] not found in tx@primary \[id\]
+INSERT INTO tx_leg VALUES (302, 3);
+
+statement ok
+COMMIT
```
Comment (dt) [inline pkg/sql/testdata/fk]: Huh, is it interesting that this is ok?

Comment (a-robinson) [inline pkg/sql/testdata/fk]: That's a good question... @knz?

**Comment by knz** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/sql/testdata/fk, line 563 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13950#-Kedy8M2QB6bFyJ91cBJ-r1-563:-Kee5fvHoFBfSfXE_On3:b-9di3as) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a7a2e7dff7d53e4e730fcad71d087003065a839/pkg/sql/testdata/fk#L563)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

That's a good question... @knz?
</blockquote></details>

It's valid in pgwire. a COMMIT while the transaction is in error state is parsed/handled as a ROLLBACK. You can check this in the interactive prompt, it reports ROLLBACK as execution tag.

A possible question is whether we want our logic test to check that the result tag of a COMMIT is indeed "COMMIT", as opposed to something else. Not sure this is warranted. Thoughts?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13950)*

**Comment by dt** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/sql/testdata/fk, line 563 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13950#-Kedy8M2QB6bFyJ91cBJ-r1-563:-Kee7mNhlGzt5a85o9rh:bt810vf) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a7a2e7dff7d53e4e730fcad71d087003065a839/pkg/sql/testdata/fk#L563)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

It's valid in pgwire. a COMMIT while the transaction is in error state is parsed/handled as a ROLLBACK. You can check this in the interactive prompt, it reports ROLLBACK as execution tag.

A possible question is whether we want our logic test to check that the result tag of a COMMIT is indeed "COMMIT", as opposed to something else. Not sure this is warranted. Thoughts?
</blockquote></details>

got it, thanks for clarifying!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13950)*

**Pull request merged**